### PR TITLE
fix: update go sdk example

### PR DESCRIPTION
**Pull request by vinckr**
Update the GO SDK documentation

Don't specify Public & Admin API directly, now point to Hydra and then call AdminApi / PublicApi methods.
Imports have changed, everything is now included in the `client` package

---

I am not sure about some things, let me know if that is still working as before in the examples:
- `NewRevokeOAuth2TokenParams()` (only found `RevokeOAuth2TokenParams`, is New needed here?) 
- `client.WithBody(&models.OAuth2Client...`
- How do I get public URL // is the "[on every request](http://localhost:3000/docs/hydra/sdk/go#on-every-request)" example still correct?

### CODE REVIEW

**Comment by gen1us2k** [comment]
@vinckr I finished with Go examples. They are correct and work fine

**Review by aeneasr — APPROVED** [review]

File: docs/hydra/sdk/go.md [hunk 1]
```diff
@@ -9,62 +9,88 @@ To install the Go SDK, run:
 go get -u -d github.com/ory/hydra-client-go
 ```
 
+:::info
+
+Be aware that endpoints `/oauth2/auth` and `/oauth2/token` **MUST NOT** be consumed
+using this SDK. Use [golang.org/x/oauth2](https://godoc.org/golang.org/x/oauth2)
+instead.
```
Comment (tomekpapiernik) [inline docs/hydra/sdk/go.md:16]: Can you give me the reason why? It'd be good to include it in the note. 
Also, there are multiple "note" formats supported by default. For a message like this one you can use
```
:::warning
```

**Comment by aeneasr** [comment]
I'd like to get this merged, how can we speed it up and bring it to a close? :)

**Comment by vinckr** [comment]
Ah sorry. I added a more clear warning as Tom suggested but I am getting some git errors

>'upstream/update-go-sdk' is not a branch

No idea why and did not have time to fix it hence not pushed yet :/. 

Tbh I also dont know why 
>`/oauth2/auth` and `/oauth2/token` **MUST NOT** be consumed
using this SDK.

I am pushing after the interview today.

**Comment by vinckr** [comment]
Can I merge this and I address Toms comment in a followup? 
I tried 20 min now to checkout this branch and cant figure it out .... 
This PR was a lot of work so would be good to merge, and I can fix the notice in a followup in a bit @aeneasr

File: docs/hydra/sdk/go.md (hunk 1 above)
Comment (aeneasr) [reply docs/hydra/sdk/go.md:16]: @vinckr you can also use suggestions inline:

````
```suggestion
Be aware that endpoints `/oauth2/auth` and `/oauth2/token` **MUST NOT** be consumed
using this SDK. Use [golang.org/x/oauth2](https://godoc.org/golang.org/x/oauth2)
instead.
```
````

**Comment by vinckr** [comment]
I tried but

```
git branch -D update-go-sdk
error: branch 'update-go-sdk' not found.
```

```
gh pr checkout 666
fatal: cannot set up tracking information; 
starting point 'upstream/update-go-sdk' is not a branch
exit status 128

```

cursed PR, never happened to me before. 😓

**Comment by vinckr** [comment]
Can I merge this now @aeneasr?
I added the notice, and we should explain why you need something else for the endpoints in https://ory.sh/docs/hydra/guides/using-oauth2 - happy to create a followup issue if you say that is an explanation that is interesting/needed for our users.

File: docs/hydra/sdk/go.md [hunk 2]
```diff
@@ -9,62 +9,88 @@ To install the Go SDK, run:
...
 
 We use code generation to generate our SDKs. The Go SDK is generated using
 [`go-swagger`](http://goswagger.io). The SDK is set up:
 
 ```go
-import "github.com/ory/hydra-client-go/client"
+package main
+
+import (
+  client "github.com/ory/hydra-client-go"
```
Comment (aeneasr) [inline docs/hydra/sdk/go.md:29]: import of `context is missing`

```suggestion
  "context"
  client "github.com/ory/hydra-client-go"
```

File: docs/hydra/sdk/go.md [hunk 3]
```diff
@@ -9,62 +9,88 @@ To install the Go SDK, run:
...
 Making requests is straight forward:
 
 ```go
+package main
+
 import (
-  "github.com/ory/hydra-client-go/client"
-  "github.com/ory/hydra-client-go/client/admin"
-  "github.com/ory/hydra-client-go/models"
+  "github.com/ory/hydra-client-go"
```
Comment (aeneasr) [inline docs/hydra/sdk/go.md:71]: `context` import missing

File: docs/hydra/sdk/go.md [hunk 4]
```diff
@@ -99,82 +134,99 @@ You may want to protect Ory Hydra using OAuth2 Access Tokens. In that case, you
...
 ### Skip TLS Verification
 
 ```go
-import "github.com/ory/hydra-client-go/client"
-import httptransport "github.com/go-openapi/runtime/client"#
-import "net/http"
+package main
+
+import (
+  "github.com/ory/hydra-client-go/client"
```
Comment (aeneasr) [inline docs/hydra/sdk/go.md:226]: Shouldn't this be

```suggestion
  "github.com/ory/hydra-client-go"
```

?

File: docs/hydra/sdk/go.md [hunk 5]
```diff
@@ -183,8 +235,14 @@ func main() {
...
   }
-  transport := httptransport.NewWithClient("host:port", "/", []string{"https"}, skipTlsClient)
-  hydra := client.New(transport, nil)
+	config := client.NewConfiguration()
+	config.Servers = []client.ServerConfiguration{
+		{
+			URL: "https://hydra.localhost:4444", // Public API URL
+		},
+	}
+	config.HTTPClient = tlsTermClient
```
Comment (aeneasr) [inline docs/hydra/sdk/go.md:244]: This variable does not exist

**Review by aeneasr — CHANGES_REQUESTED** [review]
Thank you! Unfortunately, some of the code examples do not compile when you run them as a go program. Can you please test that all examples can run 1:1 if you copy paste them? That will help developers using this guide tremendously! Otherwise they'll just say: oh I tried the docs but it sucks bcs nothing compiles

File: docs/hydra/sdk/go.md [hunk 6]
```diff
@@ -73,20 +99,29 @@ func main() {
...
-    publicURL := url.Parse("https://hydra.localhost:4444")
-    hydraPublic := hydra.NewHTTPClientWithConfig(nil, &client.TransportConfig{Schemes: []string{publicURL.Scheme}, Host: publicURL.Host, BasePath: publicURL.Path})
+  config := client.NewConfiguration()
+  config.Servers = []client.ServerConfiguration{
+    {
+      URL: "https://hydra.localhost:4444", // Public API
+    },
+  }
+
+  config.HTTPClient.Transport = httptransport.BasicAuth("my-client", "foobar")
```
[the remark this answers is not in the document]
Comment (gen1us2k) [reply docs/hydra/sdk/go.md]: Updated

File: docs/hydra/sdk/go.md (hunk 4 above)
Comment (gen1us2k) [reply docs/hydra/sdk/go.md]: I updated with correct example

**Comment by gen1us2k** [comment]
@aeneasr @vinckr Hello. So I rewrote some examples for the guide from scratch and also added useful links for additional examples. It looks good to me at the moment. Waiting for your review

**Review by vinckr — COMMENTED** [review]
Maybe we can add a short explanation as to what each code example does; it is not very obvious to me from the headline.

Some suggestions:

## Configuration

The following code example shows how to set up and configure the Ory Hydra Go SDK:

## Making requests

The following code example shows how to make requests to the Ory Hydra Public API. In this example the request is used to create an OAuth 2.0 client:

## Status codes and error handling

The following code example shows how to handle errors and status codes: 

## With Authorization

The following code example shows how to make an authenticated request to the Ory Hydra Admin API:

## Skip TLS Verification

The following code example shows how skip the TLS verification. 
(Maybe add when you want to do this, for example for testing, etc.?)

---

Also I think we can move the initial text to the top/intro

>We use the [OpenAPI specification](https://www.openapis.org/) to generate our SDKs. The Go SDK is generated using [`go-swagger`](http://goswagger.io). Read more in the [ory/sdk repository](https://github.com/ory/sdk/).

**Pull request merged**